### DISCUSSION TITLE
getDoctrineSchemaManager doesn't exist

### DISCUSSION THREAD

**Discussion by woochiwah** [reliese/laravel#283] [General]
I tried and get this message:

BadMethodCallException

  Method Illuminate\Database\MySqlConnection::getDoctrineSchemaManager does not exist.

**Comment by Wolf139** [comment]
Hi, I had the same issue as I'm using PostgreSQL. Laravel 11 no longer uses the getDoctrineSchemaManager. To fix it you have to edit the Schema.php in the reliese - laravel - sc -meta -(choose database) folder:


I replaced this:

    // public static function schemas(Connection $connection)
    // {
    //     $schemas = $connection->getDoctrineSchemaManager()->listDatabases();

    //     return array_diff($schemas, [
    //         'postgres',
    //         'template0',
    //         'template1',
    //     ]);
    // }

With this code:

use Illuminate\Support\Facades\DB;

    public static function schemas(Connection $connection)
    {
        $schemas = DB::connection($connection->getName())->table('information_schema.schemata')
            ->select('schema_name')
            ->whereNotIn('schema_name', ['pg_catalog', 'information_schema'])
            ->get()
            ->pluck('schema_name')
            ->toArray();
    
        return $schemas;
    }

**Reply by Benjaminhu** [reply]
@Wolf139 Please, if you know, take a look at my solution:  https://github.com/reliese/laravel/pull/320
